cmd/cl-adm/cmd/init/init_peer.go

```diff
@@ -0,0 +1,200 @@
+		}
+	}
+
+	return cmd
+}
+
+// AddFlags adds flags to fs and binds them to options.
+func (o *InitPeerOptions) AddFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&o.Name, "name", "", "Peer name.")
+	fs.StringVar(&o.Namespace, "namespace", app.SystemNamespace, "Namespace where the ClusterLink components are deployed.")
+	fs.StringVar(&o.CertFolder, "cert-folder", ".", "The folder where the certificates for the fabric and peer are located.")
+	fs.BoolVar(&o.RunInstance, "run", false, "If true, deploys a ClusterLink instance that will create the ClusterLink components.")
+}
+
+// RequiredFlags are the names of flags that must be explicitly specified.
+func (o *InitPeerOptions) RequiredFlags() []string {
+	return []string{"name"}
+}
+
+// Run the 'create peer' subcommand.
+func (o *InitPeerOptions) Run() error {
+	peerFolder := o.CertFolder + "/" + o.Name
+	if err := o.verifyExists(peerFolder); err != nil {
+		return err
+	}
+
+	// Create k8s resources
+	cfg, err := ctrl.GetConfig()
+	if err != nil {
+		return err
+	}
+
+	resource, err := resources.New(cfg)
+	if err != nil {
+		return err
+	}
+
+	// Create operator
+	if err := o.deployFolder("operator/manager/*", resource); err != nil {
+		return err
+	}
+
+	if err := o.deployFolder("operator/rbac/*", resource); err != nil {
+		return err
+	}
+	if err := o.deployFolder("crds/*", resource); err != nil {
+		return err
+	}
+
+	// Create cl-secret
+	secretFileName := peerFolder + "/" + config.K8SSecretYAMLFile
+	secretFile, err := os.ReadFile(secretFileName)
+	if err != nil {
+		return err
+	}
+
+	if err := o.deploySecretFile(string(secretFile), o.Namespace, resource); err != nil {
+		return err
+	}
+
+	// Create ClusterLink instance
+	if o.RunInstance {
+		instance, err := platform.K8SClusterLinkInstanceConfig(&platform.Config{
+			Peer:              o.Name,
+			Dataplanes:        1,
+			DataplaneType:     platform.DataplaneTypeEnvoy,
+			LogLevel:          "info",
+			ContainerRegistry: "ghcr.io/clusterlink-net", // Tell kind to use local image.
+			Namespace:         o.Namespace,
+			IngressType:       "NodePort",
+		}, "cl-instance")
+		if err != nil {
+			return err
+		}
+
+		err = decoder.DecodeEach(context.Background(), strings.NewReader(string(instance)), decoder.CreateHandler(resource))
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// verifyExists verifies a given path exist.
+func (o *InitPeerOptions) verifyExists(path string) error {
+	_, err := os.Stat(path)
+	return err
+}
+
+// deployFolder deploys K8s yaml from a folder.
+func (o *InitPeerOptions) deployFolder(path string, resource *resources.Resources) error {
+	err := decoder.DecodeEachFile(context.Background(), configFiles.ConfigFiles, path, decoder.CreateHandler(resource))
+	if errors.IsAlreadyExists(err) {
+		return nil
+	}
+
+	return err
+}
+
+// deploySecretFile deploys all the secret in YAML file.
+func (o *InitPeerOptions) deploySecretFile(yamlContent, newNamespace string, resource *resources.Resources) error {
```

Comment:
Why can't you simply use `decoder.DecodeEachFile`?

Reply:
Because I want to replace the namespace (without replacing just decodefile will be enough)

Reply:
You can change namespace using the decoder, e.g.:
```
func (c *KindCluster) CreateFromYAML(yaml, namespace string) error {
	return decoder.DecodeEach(context.Background(),
		strings.NewReader(yaml),
		decoder.CreateHandler(c.resources),
		decoder.MutateNamespace(namespace))
}
```